benchmarks/Balancing/dynamics.m

```diff
@@ -0,0 +1,12 @@
+function dx = dynamics(x, f)
+
+% Cartpole Swingup, 4 states x and the input f (action of the controller)
+% The controller takes (x(1), x(2), x(3), x(4)) as input, its output can
+% be used directly.
+
+dx(1,1) = x(2);
+dx(2,1) = 2*f;
+dx(3,1) = x(4);
+dx(4,1) = (0.08*0.41*(9.8*sin(x(3))-dx(2,1)*cos(x(3)))-0.0021*x(4))/0.0105;
```

Comment:
@manuelkran: 
1. The description says that x3 is the angle and x4 is the angular velocity (which is the standard state space for the cart/pole model), but here in the dynamics you swapped them. Could you please fix that?
2. I find it slightly confusing to see `dx(2,1)` on the right-hand side. I suggest to write `2*f`.
3. Do you have a reference/source for the dynamics equations? These are not exactly the equations I know.
4. Can we call the model `CartPole`, which is the common name?

Reply:
Ad 1. & 3.:
I agree to change this. I will replace dx(2,1) with 2*f in the right hand side of dx(4,1) and rename the benchmark to CartPole.

Ad 2. :
The dynamics equations describe a somewhat different, simpler CartPole system (that actually, physically exists), where the cart itself is not influenced by the pole movement.
This is due to the cart position being controlled separately, which allows us to directly set the cart acceleration.

Here is a sketch of the derivation:
  
![image](https://github.com/verivital/ARCH-COMP2024/assets/142383260/a41bc982-3d7f-4539-8fe6-ac8fa5606231)

Please let me know if something is unclear about that.